pkg/test/framework/components/namespace/kube.go

```diff
@@ -182,14 +182,24 @@ func newKube(ctx resource.Context, nsConfig *Config) (Instance, error) {
 }
 
 // createNamespaceLabels will take a namespace config and generate the proper k8s labels
-func createNamespaceLabels(cfg *Config) map[string]string {
+func createNamespaceLabels(ctx resource.Context, cfg *Config) map[string]string {
 	l := make(map[string]string)
 	l["istio-testing"] = "istio-test"
 	if cfg.Inject {
-		if cfg.Revision != "" {
-			l[label.IoIstioRev.Name] = cfg.Revision
-		} else {
-			l["istio-injection"] = "enabled"
+		// do not add namespace labels when dealing with multiple revisions since
+		// this disables the necessary object selectors
+		if !ctx.Settings().Versions.IsCrossVersion() {
+			if cfg.Revision != "" {
+				l[label.IoIstioRev.Name] = cfg.Revision
+			} else {
+				l["istio-injection"] = "enabled"
```

Comment:
We don't want to enable injection if `Revision` is specified?

Reply:
If `Revision` is specified we should add `istio.io/rev` to namespace as long as we're not running with multiple `--istio.test.versions`. Otherwise we need to leave the namespace label off so the object selectors don't get ignored